CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.13.4)
+project(CMakePublic VERSION 1.0.0 LANGUAGES CXX)
+
+add_library(CMakePublic cmake_public.cpp)
+
+#include header file
+target_include_directories (CMakePublic PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+
+install(TARGETS CMakePublic
+    EXPORT CMakePublic-targets
+    PUBLIC_HEADER DESTINATION include
+    ARCHIVE DESTINATION lib
```

Comment:
You don't want to hard code the destination like this, use [this](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) CMake module, which determines the defaults for the particular system and allows the user to override them if they wish.